### DISCUSSION TITLE
Confusion regarding backup files

### DISCUSSION THREAD

**Discussion by copax** [WinMerge/winmerge#1647] [Q&A]
In v2.16.24.0 x64 Options -> Backup Files I have my options set as below
![Options](https://user-images.githubusercontent.com/1197238/210440447-8482ce02-e0a3-4378-b9f8-bc1d2636f0e3.png)

I'm doing a Folder comparison.
I double-click a file in the Folder comparison to open the File comparison
For a desired change, I click Copy to Right
The file in the right-hand pane shows as modified.  
I click Save

My expectation is that at this point in the Right folder, a new .bak file would be created containing the contents prior to my Copy to Right. However, there is no .bak file. 

If I go back to WinMerge and click Undo, making the file in the right-hand pane modified again, and click Save, at this point the .bak file is created. 

Am I misunderstanding the intended use of the Backup Files feature? I assumed it would create a backup of my file prior to any merging, not after I undid a merge?

**Comment by sdottaka** [comment]
> Am I misunderstanding the intended use of the Backup Files feature?

I think your understanding of how Backup Files works is correct.

I have not been able to reproduce this issue.
How do you check if the backup file was created? Are you checking in Explorer or in WinMerge's folder comparison window?
If it's the latter, the .bak file won't show up unless you save and compare again.

**Reply by copax** [reply]
I have the Right side folder open in Explorer ordered by Date Modified.  I refresh before I merge and save and verify that no files have yet been modified and no new files have been created.  After I merge and save in WinMerge, I go back and refresh the Explorer window. It shows the file I merged and saved as having been modified, but there's no .bak file.  I close out of the File comparison in WinMerge, close out of the Folder comparison, and even close WinMerge itself and the .bak file never appears.  I'm wondering if I shouldn't just reinstall WinMerge at this point.

**Comment by sdottaka** [comment]
Is "Unable to backup original file ..." checked in the MessageBoxes category of the Options dialog as shown below?

If so, no notification will occur even if backup file creation fails.

![image](https://user-images.githubusercontent.com/98126/210808338-a68978d8-510e-44dc-8a84-566cebff61e5.png)

**Comment by copax** [comment]
I think I may have identified my problem. I didn't realize the bak file was being created with the same Modified Date as the original file.  I've got 6,000+ files in my folder compare, so I wasn't paying enough attention to when the file was actually being created, just when I could see it in Explorer.  Apologies for the confusion on my part.
